framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -418,16 +418,37 @@ private AnnotationMirror getAnnotationMirrorOfMetaAnnotation(
         for (Pair<AnnotationMirror, AnnotationMirror> r : declAnnotations) {
             AnnotationMirror anno = r.first;
             AnnotationMirror metaAnno = r.second;
-            List<String> expressions =
-                    AnnotationUtils.getElementValueArray(anno, "value", String.class, false);
-            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
-            if (postcondAnno == null) {
-                continue;
-            }
-            for (String expr : expressions) {
-                result.add(new Postcondition(expr, postcondAnno, anno));
+            try {
+                // If multiple contracts, then this will throw an exception which is catched at Line
+                // 433
+                List<String> expressions =
+                        AnnotationUtils.getElementValueArray(anno, "value", String.class, false);
```

Comment:
I think I understand the code now.  The `value` field is always either:
 * a list of Strings, or
 * a list of AnnotationMirrors.

Rather than using an "if" test, you attempt an assignment and that determines which case to execute.

Use of thrown and caught exceptions for control flow is confusing.
I suggest that you make an explicit test.
The test can set a variable that is a list of AnnotationMirror:  either
 * a newly-created singleton list, or
 * the list in the annotation

Then, most of the logic can be the same for both cases.

Reply:
Via reflection, [`AnnotatedElement.getAnnotationsByType(Class<T>)`](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/AnnotatedElement.html#getAnnotationsByType-java.lang.Class-) abstracts away whether repeated annotations or a wrapper were used.  Maybe there is a compile-time interface that is similar that you could use.

Reply:
@mernst I am not able to found the interface which can perform the task but I have modified the logic in a different way in #2625. Please give a look at it and review it!
Thanks.